CitadelJS/config
# added basic config files for babel, ts and webpack

This PR adds some config files for https://github.com/CitadelJS/core/issues/4.

It supports:
- build (commonjs)
- typings
- tests (JS/TS) with sourcemap support
- "merged" coverage form sources

What have to be done in @citadel/XYZ?
- the base webpack configuration should be extended to add a output filename; the created webpack configuration should be named ```webpack.config.js```
- add devDeps (can't be shared :()
- add test/index.js that looks like:
```javascript
require('source-map-support').install();
var testsContext = require.context(".", true, /\.(ts|js)$/);
testsContext.keys().forEach(testsContext);
```
- I usually add these npm scripts:
```json
{
   "scripts": {
      "watch": "webpack -w",
      "build": "webpack",
      "build-tests": "webpack --config webpack-test.config.js",
      "test": "npm run build-tests && istanbul cover --print none --report json _mocha -- --full-trace generated/tests.js && remap-istanbul -i coverage/coverage.json > coverage/coverage-final.json && node scripts/generate-coverage.js text",
      "coverage-html": "npm run test && node scripts/generate-coverage.js html",
      "coverage-lcovonly": "npm run test && node scripts/generate-coverage.js lcovonly"
   }
}
```
